### PR TITLE
Place order endpoint

### PR DESCRIPTION
**Pull request by simionescuraluca**

### CODE REVIEW

File: src/test/java/com/vinyl/PlaceOrderTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    @Test
+    public void testWhenStockIsTooSmall() {
+        product.setStock(0);
+        productRepository.save(product);
+
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
```
Comment (dragosslujeru) [inline src/test/java/com/vinyl/PlaceOrderTest.java:66]: also assert that purchase is not created

File: src/test/java/com/vinyl/PlaceOrderTest.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        product = defaultEntitiesHelper.createProduct();
+        cart = cartSetup();
+    }
+
+    @Test
+    public void testWhenOk() {
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        Assertions.assertThat(purchaseRepository.findByUser(user)).isNotNull();
```
Comment (dragosslujeru) [inline src/test/java/com/vinyl/PlaceOrderTest.java]: also assert that the product from the purchase is the correct one

File: src/test/java/com/vinyl/PlaceOrderTest.java (hunk 1 above)
Comment (simionescuraluca) [reply src/test/java/com/vinyl/PlaceOrderTest.java:66]: done

File: src/test/java/com/vinyl/PlaceOrderTest.java (hunk 2 above)
Comment (simionescuraluca) [reply src/test/java/com/vinyl/PlaceOrderTest.java]: done

File: src/test/java/com/vinyl/PlaceOrderTest.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    @Test
+    public void testWhenOk() {
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        Purchase purchase = purchaseRepository.findByUser(user);
+        Assertions.assertThat(purchase).isNotNull();
+        Assertions.assertThat(purchaseProductRepository.findByPurchase(purchase).getProduct().getProductName()).isEqualTo(product.getProductName());
```
Comment (dragosslujeru) [inline src/test/java/com/vinyl/PlaceOrderTest.java:48]: no need to delcare a new method in the repo for this assert, find the purchase by user and check the first product from the product list

**Review by dragosslujeru — APPROVED** [review]

**Pull request merged**